.claude/skills/productize-chart-env/scripts/gen-schema.py

```
@@ -350,11 +359,27 @@ def main():
     values = yaml.safe_load(open(a.values)) or {}
     desc = descriptions(a.values)
     tgroups = template_group_fields(a.chart_dir) if a.chart_dir else {}
+    # Auto-detect this chart's subcharts from Chart.yaml so their (opaque, upstream)
+    # value blocks stay permissive — the hardcoded SUBCHARTS set can't know every
+    # dependency name (e.g. seaweedfs, keda, otel-collector-lerian). Union, never shrink.
+    if a.chart_dir:
+        cy = os.path.join(a.chart_dir, "Chart.yaml")
+        if os.path.exists(cy):
+            meta = yaml.safe_load(open(cy)) or {}
```

Ruff (0.16.1)

[warning] Use a context manager for opening files

(SIM115)